azurerm/resource_arm_analysis_services_server_test.go

```diff
@@ -0,0 +1,409 @@
+					testCheckAzureRMAnalysisServicesServerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "querypool_connection_mode", "ReadOnly"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMAnalysisServicesServer_firewallSettings(t *testing.T) {
+	resourceName := "azurerm_analysis_services_server.test"
+	ri := tf.AccRandTimeInt()
+
+	config1 := testAccAzureRMAnalysisServicesServer_firewallSettings1(ri, testLocation(), true)
+
+	config2 := testAccAzureRMAnalysisServicesServer_firewallSettings2(ri, testLocation(), false)
+
+	config3 := testAccAzureRMAnalysisServicesServer_firewallSettings3(ri, testLocation(), true)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMAnalysisServicesServerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config1,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAnalysisServicesServerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "enable_power_bi_service", "true"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.#", "0"),
+				),
+			},
+			{
+				Config: config2,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAnalysisServicesServerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "enable_power_bi_service", "false"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.name", "test1"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.range_start", "92.123.234.11"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.range_end", "92.123.234.12"),
+				),
+			},
+			{
+				Config: config3,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAnalysisServicesServerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "enable_power_bi_service", "true"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.name", "test1"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.range_start", "92.123.234.11"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.range_end", "92.123.234.13"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.1.name", "test2"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.1.range_start", "226.202.187.57"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.1.range_end", "226.208.192.47"),
+				),
+			},
+		},
+	})
+}
+
+// ARM_ACC_EMAIL1 and ARM_ACC_EMAIL2 must be set and existing emails in the tenant's AD to work properly
+func TestAccAzureRMAnalysisServicesServer_adminUsers(t *testing.T) {
+	const ArmAccEmail1 = "ARM_ACC_EMAIL1"
+	const ArmAccEmail2 = "ARM_ACC_EMAIL2"
```

Comment:
could we make these keys more descriptive? perhaps `ARM_ACCTEST_ADMIN_EMAIL1` and `ARM_ACCTEST_ADMIN_EMAIL2`?